### PR TITLE
Feature/us787219 sample UI test case api geo location

### PR DESCRIPTION
**Pull request by pratap511**

### CODE REVIEW

File: MASStockTrading.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -211,18 +276,20 @@
...
 			name = "[CP] Embed Pods Frameworks";
 			outputPaths = (
+				"${TARGET_BUILD_DIR}/${FRAMEWORKS_FOLDER_PATH}/MASProximity.framework",
 				"${TARGET_BUILD_DIR}/${FRAMEWORKS_FOLDER_PATH}/MASFoundation.framework",
 				"${TARGET_BUILD_DIR}/${FRAMEWORKS_FOLDER_PATH}/MASUI.framework",
 			);
 			runOnlyForDeploymentPostprocessing = 0;
 			shellPath = /bin/sh;
-			shellScript = "\"${SRCROOT}/Pods/Target Support Files/Pods-MASStockTrading/Pods-MASStockTrading-frameworks.sh\"\n";
+			shellScript = "\"${PODS_ROOT}/Target Support Files/Pods-MASStockTrading/Pods-MASStockTrading-frameworks.sh\"\n";
```
Comment (ysd24) [inline MASStockTrading.xcodeproj/project.pbxproj]: Do not checkin the xcworkspace changes created by cocoapods. Please revert back to old state. Add your UITests files and checkin only the relevant xcodeproj/pbxproj changes.

File: MASStockTrading.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (ysd24) [inline MASStockTrading.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist]: Do not checkin this file.

File: MASStockTrading/Base.lproj/Main.storyboard [hunk 3]
```diff
@@ -1,8 +1,9 @@
-<?xml version="1.0" encoding="UTF-8" standalone="no"?>
-<document type="com.apple.InterfaceBuilder3.CocoaTouch.Storyboard.XIB" version="3.0" toolsVersion="11201" systemVersion="15G1004" targetRuntime="iOS.CocoaTouch" propertyAccessControl="none" useAutolayout="YES" colorMatched="YES" initialViewController="dl0-ky-aLA">
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (ysd24) [inline MASStockTrading/Base.lproj/Main.storyboard]: Do you require these changes to the Main.Storyboard ? Please revert these changes if they are not intentionally changed !!

File: MASStockTrading/Classes/MASMainViewController.m [hunk 4]
```diff
@@ -62,6 +62,8 @@ - (void)viewDidLoad
 //        
 //    }];
     
+    [MAS setGatewayNetworkActivityLogging:YES];
```
Comment (ysd24) [inline MASStockTrading/Classes/MASMainViewController.m]: Revert if no more needed !! Do not checkin any lines added for debugging purpose !!

File: MASStockTradingUITests/MASStockTradingUITests.m [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+#define TIME_INTERVAL                     10
+#define MAS_UI_CANCEL                     @"masui-cancelBtn"
+#define MAS_UI_LogIn                      @"Login"
+#define MAS_UI_USER_TEXTFIELD             @"masui-usernameField"
+#define MAS_UI_PASSWORD_FIELD             @"masui-passwordField"
+#define USER_CANCEL_OTP_SELECTION         @"OTP channel selection has been cancelled by user."
+#define MAS_RESPONSE_INFO_BODY_INFO_KEY   @"MASResponseInfoBodyInfoKey"
+#define MAS_RESPONSE_INFO_HEADER_INFO_KEY @"MASResponseInfoHeaderInfoKey"
+#define INVALID_CREDENTIALS               @"Username or password invalid"
+#define OK_BUTTON                         @"OK"
```
Comment (ysd24) [inline MASStockTradingUITests/MASStockTradingUITests.m:32]: Please maintain a separate json file and take input from there. Also if possible don't expose any sensitive data like username/passwords as these sample apps will get shared with customers too.

File: MASStockTradingUITests/MASStockTradingUITests.m [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+
+            return true;
+        }
+        return false;
+    }
+    return false;
+}
+
+- (void)test_BUY_Stocks {
+    [_app launch];
```
Comment (ysd24) [inline MASStockTradingUITests/MASStockTradingUITests.m:128]: Please add proper comments - There are not comments at all. We can not make out what each step is trying to do.
Also please follow our commenting style for example :
//
// Invoke protected endpoint
//

File: MASStockTradingUITests/MASStockTradingUITests.m [hunk 7]
```diff
@@ -0,0 +1,220 @@
...
+        
+    } else {
+        
+        [self waitForElement:textViewElement];
+        NSString *strResult=[NSString stringWithFormat:@"%@",textViewElement.value];
+        XCTAssert([strResult containsString:MAS_RESPONSE_INFO_BODY_INFO_KEY]);
+        XCTAssert([strResult containsString:MAS_RESPONSE_INFO_HEADER_INFO_KEY]);
+    }
+}
+
```
Comment (ysd24) [inline MASStockTradingUITests/MASStockTradingUITests.m:227]: Please add negative test scenarios. I don't see an invalid case where you try to buy/sell a large number of shares say 10000 and above, a OTP screen is shown. You can for now cancel the OTP request and pass the test as negative test scenario.

**Review by ysd24 — COMMENTED** [review]
Added review comments.

File: MASStockTrading.xcodeproj/project.pbxproj (hunk 1 above)
Comment (pratap511) [reply MASStockTrading.xcodeproj/project.pbxproj]: @ysd24  Reverted these file changes

File: MASStockTrading.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist (hunk 2 above)
Comment (pratap511) [reply MASStockTrading.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist]: @ysd24 Reverted these file changes

File: MASStockTrading/Base.lproj/Main.storyboard (hunk 3 above)
Comment (pratap511) [reply MASStockTrading/Base.lproj/Main.storyboard]: @ysd24 reverted these changes

File: MASStockTrading/Classes/MASMainViewController.m (hunk 4 above)
Comment (pratap511) [reply MASStockTrading/Classes/MASMainViewController.m]: @ysd24 changed accordingly

File: MASStockTradingUITests/MASStockTradingUITests.m (hunk 5 above)
Comment (pratap511) [reply MASStockTradingUITests/MASStockTradingUITests.m:32]: @ysd24 created separate JSON File and handling data in that file

File: MASStockTradingUITests/MASStockTradingUITests.m (hunk 6 above)
Comment (pratap511) [reply MASStockTradingUITests/MASStockTradingUITests.m:128]: @ysd24 provided comments as per changes

File: MASStockTradingUITests/MASStockTradingUITests.m (hunk 7 above)
Comment (pratap511) [reply MASStockTradingUITests/MASStockTradingUITests.m:227]: @ysd24 added assertions. When cancel button press, on result text view handling cancel text.

**Review by ysd24 — APPROVED** [review]
Looks good.
